### ISSUE TITLE
Don't fail silently if sign_transaction fails

### ISSUE THREAD

**Issue by theoreticalbts** [steemit/tinman#61]
Currently the `sign_transaction` issue #60 is failing.  I think ignoring this was intentional, since ported transactions can fail for a variety of reasons.  But perhaps this should be revisited.

**AC**
- [ ] Decided which way we want to go, per [this comment](https://github.com/steemit/tinman/issues/61#issuecomment-407552959) Put results in `# Results` comment.

**Comment by inertia186** [comment]
**a)** Keep the `-die` option the way it is and require the user to pass the argument if they want this behavior.

**b)** Change `die_on_fail` default to `True` and require the user to pass something like a `-force` argument to intentionally ignore transaction failure.

**Comment by mvandeberg** [comment]
Signing and pushing a transaction are two distinct steps in tinman that I am not sure this issue captures the nuance of.

Signing a transaction should never fail in tinman. So long as it has a transaction and a key, it should pass. The sign transaction binary takes a private key and a transaction and returns a signed transaction with the signature added to the transaction (can be used for incremental multisig).

Pushing the transaction can fail for a variety of reasons and should be ignored. Push transaction accepts a signed transaction, applying to the blockchain and broadcasting the peers if successful. Otherwise returning an error, which can be ignored.

**Comment by mvandeberg** [comment]
In the case of #60, the failures were logged to Scalyr and we were able to diagnose the problem. It didn't fail silently. I do not see a problem with the current approach.

**Comment by theoreticalbts** [comment]
I agree.  If the `sign_transaction` process fails, it means there's a serious bug in `tinman` or `sign_transaction` (e.g. producing malformed JSON, or generating a transaction that the C++ code doesn't understand how to serialize).  The `tinman` process should immediately exit with a nonzero error code in this case, regardless of its configuration.

Pushing a transaction can fail for a variety of reasons.  The main reason is that we don't expect the chain state to be 100% identical on testnet vs mainnet, so pushing some transactions ported from mainnet should be expected to fail.

**Comment by inertia186** [comment]
Should we whitelist certain error responses and `exit(-1)`, like:

* `Input data have to treated as object.`
* `transaction expiration exception`
* `Duplicate transaction check failed`
* `signature is not canonical`
* `attempting to push a block that is too old`
* `irrelevant signature`
* `missing required posting authority`
* `missing required active authority`
* `missing required owner authority`
* `missing required other authority`
* `Invalid operation name`
* `Cannot transfer a negative amount (aka: stealing)`
* `Must transfer a nonzero amount`
* `is_asset_type`

Basically, if you saw these, wouldn't you want it to fail quickly?

**Comment by inertia186** [comment]
Re-open if anyone is interested in fail-fast.
